hazelcast/src/main/java/com/hazelcast/map/impl/operation/PostJoinMapOperation.java

```diff
@@ -190,9 +191,9 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
     protected void readInternal(ObjectDataInput in) throws IOException {
         super.readInternal(in);
         // RU_COMPAT_39
-        // Version 3.9.x still sends index info's because PostJoinMapOperation was not marked Versioned
+        // Version 3.9.x still sends index info's from EE because PostJoinMapOperation was not marked Versioned
         Version inputversion = in.getVersion();
-        if (inputversion.isUnknown() || inputversion.isEqualTo(V3_9)) {
+        if ((inputversion.isUnknown() || inputversion.isEqualTo(V3_9)) && getBuildInfo().isEnterprise()) {
```

Comment:
Is this the reason for the OOM in the RU test?
(and I think I also saw it in a real-life test from Danny)

Reply:
Yes, this usually manifests either as OOME (I think on the 3.9.x side) or as malformed bytes sequence when attempting to read UTF string.